phpBB/assets/javascript/installer.js

```diff
@@ -0,0 +1,221 @@
+        var logContainer = $('#log-container');
+
+        var title, description, msgElement, arraySize = messages.length;
+        for (var i = 0; i < arraySize; i++)
+        {
+            msgElement = $(document.createElement('div'));
+            title = $(document.createElement('strong'));
+            title.text(messages[i].title);
+            msgElement.append(title);
+
+            if (messages[i].hasOwnProperty('description'))
+            {
+                description = $(document.createElement('p'));
+                description.text(messages[i].description);
+                msgElement.append(description);
+            }
+
+            switch (type)
+            {
+                case 'error':
+                    msgElement.addClass('errorbox');
+                    errorContainer.append(msgElement);
+                break;
+                case 'warning':
+                    msgElement.addClass('warningbox');
+                    warningContainer.append(msgElement);
+                break;
+                case 'log':
+                    msgElement.addClass('log');
+                    logContainer.append(msgElement);
+                break;
+            }
+        }
+    }
+
+    function add_form(formHtml)
+    {
+        var formContainer = $('#content-container');
+        formContainer.html(formHtml);
+        var form = $('#install_install');
+        intercept_form_submit(form);
+    }
+
+    function start_polling(xhReq)
+    {
+        reset_polling();
+        pollTimer = setInterval(function() { poll_content(xhReq); }, 500)
+    }
+
+    function reset_polling()
+    {
+        clearInterval(pollTimer);
+        nextReadPosition = 0;
+    }
+
+    function submit_form(form, submitBtn)
+    {
+        form.css('display', 'none');
+
+        var xhReq = create_xhr_object();
+        xhReq.open("POST", form.attr('action'), true);
+        xhReq.setRequestHeader('X-Requested-With', 'XMLHttpRequest');
+        xhReq.setRequestHeader('Content-type', 'application/x-www-form-urlencoded');
+        xhReq.send(get_form_fields(form, submitBtn));
+
+        // Clear content
+        setup_ajax_layout();
+        $('#loading_indicator').css('display', 'block');
+
+        start_polling(xhReq);
+    }
+
+    // Workaround for submit buttons
+    function get_form_fields(form, submitBtn)
+    {
+        var formData = form.serialize();
+        //var submitBtn = form.find(':submit');
+        formData += ((formData.length) ? '&' : '') + encodeURIComponent(submitBtn.attr('name')) + '=';
+        formData += encodeURIComponent(submitBtn.attr('value'));
+
+        return formData;
+    }
+
+    function intercept_form_submit(form)
+    {
+        if (!form.length)
+        {
+            return;
+        }
+
+        form.find(':submit').bind('click', function(event) {
+            event.preventDefault();
+            submit_form(form, $(this));
+        });
+
+    }
+
+    function create_xhr_object()
```

Comment:
I have no idea how to proccess streamed response with jQuery, AFAIK it is impossible to access response text while connection is still alive.